src/coreclr/vm/jitinterface.cpp

```diff
@@ -1561,7 +1561,33 @@ void CEEInfo::getFieldInfo (CORINFO_RESOLVED_TOKEN * pResolvedToken,
                 GCX_COOP();
 
                 pResult->fieldLookup.addr = pField->GetStaticAddressHandle((void*)pField->GetBase());
-                pResult->fieldLookup.accessType = IAT_VALUE;
+                GCPROTECT_BEGININTERIOR(pResult->fieldLookup.addr);
```

Comment:
What guarantees that `pResult->fieldLookup.addr` is not going to move once it is return to the JIT?

Reply:
@jkotas I don't have a good explanation yet, what I understand so far:
1) `GetGCStaticsBasePointer` is assumed to be moveable already: https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/methodtable.cpp#L3485
2) I thought that it supposed to return an address to a pinned area so that's how we can give it to JIT
but apparently it works differently for collectible contexts? And in case of collectible context is not supposed to be consumed by JIT?
3) the places where it's used already seem to be safe? because of COOP and nothing triggers GC (and e.g. BulkStaticsLogger::LogAllStatics() has GC_NOTRIGGER)

Reply:
Yes, the statics for collectible types do not seem to be pinned: https://github.com/dotnet/runtime/blob/1ad1022083def722cf43f057a46b644753543165/src/coreclr/vm/appdomain.cpp#L4184 . I assume that it is to prevent GC heap fragmentation that the long-term pinning would cause.

I think that this means that we need to avoid giving the JIT address of the collectible type statics?

Reply:
> I think that this means that we need to avoid giving the JIT address of the collectible type statics?

[Looks like we already don't do it](https://github.com/dotnet/runtime/blob/a83c661bb8061b65404fb154b328ff1d3993d9ec/src/coreclr/vm/jitinterface.cpp#L1537-L1546), at least we don't hit that path where we set the address.